### PR TITLE
Refactor star_targets rules

### LINKED ISSUE #47: Add actions to data/simpy.gram
There's a complete Python grammar in `data/simpy.gram`, but it is lacking actions.

OTOH there are actions (written in C) for a very small subset in `data/cprog.gram` -- these are tested by the `Makefile` targets `test`, `compile`, `dump` and `time`. The actions construct accurate AST nodes.

We should add actions to `simpy.gram` modeled after those in `cprog.gram`.

This will likely require refactoring the grammar in `simpy.gram`.

### LINKED ISSUE #124: Simpy refactorings: star_targets

### PR DESCRIPTION
**Pull request by lysnikolaou**
Closes #124.

I maybe rushing this a bit but doesn't this also close #47? (not considering f-strings)

### CODE REVIEW

File: test/test_ast_generation.py [hunk 1]
```diff
@@ -119,6 +119,13 @@ def f():
         else:
             pass
      '''),
+    ('for_star_targets_attribute', 'for a.b in c: pass'),
+    ('for_star_targets_call_attribute', 'for a().c in b: pass'),
+    ('for_star_targets_mixed', 'for a[0].b().c in d: pass'),
+    ('for_star_targets_multiple', 'for a, b in c: pass'),
+    ('for_star_targets_starred', 'for *a in b: pass'),
+    ('for_star_targets_subscript_attribute', 'for a[0].b in c: pass'),
+    ('for_star_targets_tuple', 'for a, (b, c) in d: pass'),
```
Comment (pablogsal) [inline test/test_ast_generation.py]: I would suggest adding (just to cover more paths):

``` 
for a, *b, (c, d) in d: pass
```

File: test/test_ast_generation.py [hunk 2]
```diff
@@ -119,6 +119,18 @@ def f():
...
+    ('for_star_targets_mixed', 'for a[0].b().c in d: pass'),
+    ('for_star_targets_mixed_starred',
+     '''
+        for a, *b, (c, d) in e:
+            pass
+     '''),
+    ('for_star_targets_multiple', 'for a, b in c: pass'),
+    ('for_star_targets_starred', 'for *a in b: pass'),
+    ('for_star_targets_subscript_attribute', 'for a[0].b in c: pass'),
+    ('for_star_targets_tuple', 'for a, (b, c) in d: pass'),
```
Comment (pablogsal) [inline test/test_ast_generation.py]: I think we are missing cases with trailing commas

**Comment by lysnikolaou** [comment]
@gvanrossum Should I wait for your review as well or merge this?

File: data/simpy.gram [hunk 3]
```diff
@@ -422,9 +422,21 @@ kwarg[KeywordOrStarred*]:
 # NOTE: star_targets may contain *NAME, targets may not.
```
Comment (gvanrossum) [inline data/simpy.gram]: It's really `*bitwise_or`.

File: data/simpy.gram [hunk 4]
```diff
@@ -422,9 +422,21 @@ kwarg[KeywordOrStarred*]:
...
 # NOTE: the !'in' is to handle "for x, in ...".
 # TODO: things like {k: v}[k] = v should also be acceptable [star] targets.
```
Comment (gvanrossum) [inline data/simpy.gram]: This is done, right?

**Comment by lysnikolaou** [comment]
Yeah, #47 is closed! 🎉 🎉 🎉

**Pull request merged**